### DISCUSSION TITLE
Esp32-cam get off led flashing and using gpio available

### DISCUSSION THREAD

**Discussion by Garycooper07** [arendst/Tasmota#11304] [Q&A] [answered]
Hi, I flashed several Esp32-cam but now I would do most of thing whith it.
First, I would to put off the flashing white led (I don't use micro SD)
Then, I would using GPIO16, GPIO1 and GPIO3 with a switch (to do video doorbell for example) or with DS18b20 or DHT22.
Also, I would to use this with domoticz but I don't find how to do that on Esp32cam.
Thanks

**Comment by ascillato** [comment, marked as answer]
Please, address this to the [Tasmota Support Chat](https://discord.gg/Ks2Kzd4). The chat is a better and more dynamic channel for helping you. 

Please check the [Contributing Guideline and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md)

Thanks.

---------------------------

### Support Information 
See [Docs](https://tasmota.github.io/docs/) for more information.
See [Chat](https://discord.gg/Ks2Kzd4) for more user experience.
See [Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)

**Comment by barbudor** [comment]
If you have already flashed Tasmota on the esp32cam board then there is nothing specific.

However build a tasmota binary with espcam and Sensors is not trivial and you need to go through the [Compile your build](https://tasmota.github.io/docs/Compile-your-build/) process and [Compilig for ESP32](https://tasmota.github.io/docs/ESP32/#compiling-for-esp32)

Note that the video is very hungry and is already taking major resources on the espcam build so you should not expect to be able to add many features, although a few I2C sensors should be ok.

Regarding controlling the led which is on GPIO4, configuring this pin as a Relay would allow you to toggle it on and off. Or better configure it as a PWM output to be able to adjust the level of light.

**Reply by Garycooper07** [reply]
Ok, thanks for your answer

**Comment by Jason2866** [comment]
To do a customized Tasmota32 Webcam build, you have to use Tasmota as base variant.
In your `user_config_override.h` you have to undefine as max. as possible features you do NOT need and define the features you want.
See version `lite` as reference what to do https://github.com/arendst/Tasmota/blob/development/tasmota/tasmota_configurations.h#L614

You have to add
```
#ifdef FIRMWARE_CAM
#define USE_WEBCAM

#undef xxx

#define yyy

#endif
```
in `user_config_override.h`

Do not forget to rename `platformio_override_sample.ini` to `platformio_override.ini` !!

Next you have to define a custom [env] for your custom webcam build best placed in new generated file called `platformio_tasmota_cenv.ini`
```
[env:tasmota32-camcustom]
extends                 = env:tasmota32
board                   = esp32cam
board_build.f_cpu       = 240000000L
board_build.flash_mode  = qio
board_build.f_flash     = 80000000L
build_flags             = ${common32.build_flags} -DBOARD_HAS_PSRAM -mfix-esp32-psram-cache-issue -lc-psram-workaround -lm-psram-workaround -DFIRMWARE_CAM
````

To compile your build run command `pio run -e tasmota32-camcustom` from VSC Terminal window

This is the way done as @barbudor linked to the [docs](https://tasmota.github.io/docs/Compile-your-build/#defining-multiple-custom-firmwares)

**Reply by Garycooper07** [reply]
Hi, thanks for you answer. I'm a noob on compiling and platformio is really difficult for noobs...
I flashed it with esptool on raspberry pi 4.

**Reply by joba-1** [reply]
try to set up a cross compilation environment yourself and you have a new definition for difficult :)
